### DISCUSSION TITLE
Invalid Signature & old PKs

### DISCUSSION THREAD

**Discussion by bernhard-ma-adm** [Budibase/budibase#8690] [Help]
Hi,
i had to change some columns (including PrimaryKey) etc. in my external Postgre-DB.
Now i see two PKs for some tables in BudiBase (the new fachid and the old fach_id). Fetching tables didn't solve this.
![image](https://user-images.githubusercontent.com/106738321/201967720-a02b16c6-8189-4cb1-96d4-01b48961490d.png)

And, in custom Querys i get alwas error `Query Error: invalid signature` - in existing and new created.

The internal BudiBase relations are up to date.
I know i have to update some of my screens, this will i do later, when these errors are solved...

What can i do to solve the problems?
Thanks!

**Comment by melohagan** [comment]
Hey @bernhard-mischokacademy 

Perhaps try removing the datasource entirely, re-add, then re-fetch tables?
